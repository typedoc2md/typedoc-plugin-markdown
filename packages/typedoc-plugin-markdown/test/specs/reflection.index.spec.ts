import { expectFileToEqual } from '@devtools/testing';

describe(`Reflection Indexes`, () => {
  test(`should compile reflection indexes`, () => {
    expectFileToEqual('reflections', 'members', ['globals.md', 'README.md']);
  });
});
