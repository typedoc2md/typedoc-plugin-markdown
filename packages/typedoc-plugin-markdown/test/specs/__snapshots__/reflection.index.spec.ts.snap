// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reflection Indexes should compile reflection indexes: (Output File Strategy "members") (Option Group "1") 1`] = `
"# typedoc-stubs

Module comments

## Enumerations

- [BasicEnum](enumerations/BasicEnum.md)
- [EnumWithValues](enumerations/EnumWithValues.md)

## Classes

- [\\_\\_DuplicateClass\\_\\_](classes/DuplicateClass.md)
- [\\_DuplicateClass\\_](classes/DuplicateClass-1.md)
- [AbstractClass](classes/AbstractClass.md)
- [BaseClass](classes/BaseClass.md)
- [BasicClass](classes/BasicClass.md)
- [CallbacksOptions](classes/CallbacksOptions.md)
- [ChildClassA](classes/ChildClassA.md)
- [ChildClassB](classes/ChildClassB.md)
- [ClassWithAccessorKeywords](classes/ClassWithAccessorKeywords.md)
- [ClassWithAccessors](classes/ClassWithAccessors.md)
- [ClassWithComplexProps](classes/ClassWithComplexProps.md)
- [ClassWithConstructorOverloads](classes/ClassWithConstructorOverloads.md)
- [ClassWithFlags](classes/ClassWithFlags.md)
- [ClassWithModifiers](classes/ClassWithModifiers.md)
- [ClassWithoutPropCategories](classes/ClassWithoutPropCategories.md)
- [ClassWithPropCategories](classes/ClassWithPropCategories.md)
- [ClassWithSimpleProps](classes/ClassWithSimpleProps.md)
- [ClassWithSymbols](classes/ClassWithSymbols.md)
- [ClassWithTypeParameters](classes/ClassWithTypeParameters.md)
- [DerivedClassA](classes/DerivedClassA.md)
- [DerivedClassB](classes/DerivedClassB.md)
- [DisposableClass](classes/DisposableClass.md)
- [DuplicateClass](classes/DuplicateClass-2.md)
- [GrandChildClassA](classes/GrandChildClassA.md)
- [GrandChildClassB](classes/GrandChildClassB.md)

## Interfaces

- [BasicInterface](interfaces/BasicInterface.md)
- [CustomEventInterface](interfaces/CustomEventInterface.md)
- [ExtendedInterface](interfaces/ExtendedInterface.md)
- [IndexableInterface](interfaces/IndexableInterface.md)
- [InterfaceWithComments](interfaces/InterfaceWithComments.md)
- [InterfaceWithEventProperties](interfaces/InterfaceWithEventProperties.md)
- [InterfaceWithFlags](interfaces/InterfaceWithFlags.md)
- [InterfaceWithTypeParameters](interfaces/InterfaceWithTypeParameters.md)
- [MultipleIndexableInterface](interfaces/MultipleIndexableInterface.md)

## Type Aliases

- [\\_\\_TypeDeclarationWithSpecialCharacters\\_](type-aliases/TypeDeclarationWithSpecialCharacters.md)
- [ArrayOfObjectsType](type-aliases/ArrayOfObjectsType.md)
- [ArrayType](type-aliases/ArrayType.md)
- [ConditionalType](type-aliases/ConditionalType.md)
- [ExplicitTypeDefOptions](type-aliases/ExplicitTypeDefOptions.md)
- [ExternalReferenceType](type-aliases/ExternalReferenceType.md)
- [FunctionType](type-aliases/FunctionType.md)
- [IndexAccessType](type-aliases/IndexAccessType.md)
- [IntersectionType](type-aliases/IntersectionType.md)
- [LiteralType](type-aliases/LiteralType.md)
- [PartialMappedType](type-aliases/PartialMappedType.md)
- [PrimitiveType](type-aliases/PrimitiveType.md)
- [PromiseTypeWithObject](type-aliases/PromiseTypeWithObject.md)
- [PromiseTypeWithSymbol](type-aliases/PromiseTypeWithSymbol.md)
- [QueryType](type-aliases/QueryType.md)
- [ReadonlyMappedType](type-aliases/ReadonlyMappedType.md)
- [StringLiteralType](type-aliases/StringLiteralType.md)
- [Stuff](type-aliases/Stuff.md)
- [TupleType](type-aliases/TupleType.md)
- [TypeDefOptions](type-aliases/TypeDefOptions.md)
- [TypeWithExternalSymbolLinkMapping](type-aliases/TypeWithExternalSymbolLinkMapping.md)
- [TypeWithReturns](type-aliases/TypeWithReturns.md)
- [TypeWithTypeParams](type-aliases/TypeWithTypeParams.md)
- [UnionType](type-aliases/UnionType.md)
- [UnionTypeWithTemplateStrings](type-aliases/UnionTypeWithTemplateStrings.md)
- [UsefulUnionType](type-aliases/UsefulUnionType.md)

## Variables

- [objectLiteralVariable](variables/objectLiteralVariable.md)
- [objectWithSymbol](variables/objectWithSymbol.md)
- [someQuery](variables/someQuery.md)
- [stringVariable](variables/stringVariable.md)
- [templateVariableString](variables/templateVariableString.md)
- [typedef](variables/typedef.md)
- [typeOperatorVariable](variables/typeOperatorVariable.md)

## Functions

- [basicFunction](functions/basicFunction.md)
- [basicFunctionWithReturns](functions/basicFunctionWithReturns.md)
- [curriedFunction](functions/curriedFunction.md)
- [functionReturningAFunction](functions/functionReturningAFunction.md)
- [functionReturningAnObject](functions/functionReturningAnObject.md)
- [functionReturningAPromise](functions/functionReturningAPromise.md)
- [functionReturningAString](functions/functionReturningAString.md)
- [functionReturningAUnionType](functions/functionReturningAUnionType.md)
- [functionWithArrayOfOptionalStuff](functions/functionWithArrayOfOptionalStuff.md)
- [functionWithArrayOfStuff](functions/functionWithArrayOfStuff.md)
- [functionWithArrayOfUnionStuff](functions/functionWithArrayOfUnionStuff.md)
- [functionWithComplexParams](functions/functionWithComplexParams.md)
- [functionWithDefaultParameters](functions/functionWithDefaultParameters.md)
- [functionWithMultipleSignatures](functions/functionWithMultipleSignatures.md)
- [functionWithNamedParams](functions/functionWithNamedParams.md)
- [functionWithNestedParameters](functions/functionWithNestedParameters.md)
- [functionWithOptionalParameters](functions/functionWithOptionalParameters.md)
- [functionWithRestParams](functions/functionWithRestParams.md)
- [functionWithTypeParameters](functions/functionWithTypeParameters.md)
- [functionWithUnionParams](functions/functionWithUnionParams.md)
- [tupleTypeFunction](functions/tupleTypeFunction.md)
- [tupleTypeFunctionInlineExplicit](functions/tupleTypeFunctionInlineExplicit.md)
- [tupleTypeFunctionOptional](functions/tupleTypeFunctionOptional.md)
"
`;

exports[`Reflection Indexes should compile reflection indexes: (Output File Strategy "members") (Option Group "2") 1`] = `
"# typedoc-stubs

Module comments

## Enumerations

| Enumeration | Description |
| :------ | :------ |
| [BasicEnum](enumerations/BasicEnum.md) | Comments for enum |
| [EnumWithValues](enumerations/EnumWithValues.md) | Comments for enum |

## Classes

| Class | Description |
| :------ | :------ |
| [\\_\\_DuplicateClass\\_\\_](classes/DuplicateClass.md) | Comments for \`__DuplicateClass__\` |
| [\\_DuplicateClass\\_](classes/DuplicateClass-1.md) | Comments for \`_DuplicateClass_\` |
| [AbstractClass](classes/AbstractClass.md) | - |
| [BaseClass](classes/BaseClass.md) | - |
| [BasicClass](classes/BasicClass.md) | Comments for BasicClass |
| [CallbacksOptions](classes/CallbacksOptions.md) | - |
| [ChildClassA](classes/ChildClassA.md) | - |
| [ChildClassB](classes/ChildClassB.md) | - |
| [ClassWithAccessorKeywords](classes/ClassWithAccessorKeywords.md) | - |
| [ClassWithAccessors](classes/ClassWithAccessors.md) | Comments for ClassWithAccessors |
| [ClassWithComplexProps](classes/ClassWithComplexProps.md) | Comments for ClassWithComplexProps |
| [ClassWithConstructorOverloads](classes/ClassWithConstructorOverloads.md) | Comments for ClassWithConstructorOverloads |
| [ClassWithFlags](classes/ClassWithFlags.md) | Comment for ClassWithFlags |
| [ClassWithModifiers](classes/ClassWithModifiers.md) | Comments for ClassWithModifiers |
| [ClassWithoutPropCategories](classes/ClassWithoutPropCategories.md) | - |
| [ClassWithPropCategories](classes/ClassWithPropCategories.md) | - |
| [ClassWithSimpleProps](classes/ClassWithSimpleProps.md) | Comments for ClassWithSimpleProps |
| [ClassWithSymbols](classes/ClassWithSymbols.md) | - |
| [ClassWithTypeParameters](classes/ClassWithTypeParameters.md) | Comments for ClassWithTypeParameters |
| [DerivedClassA](classes/DerivedClassA.md) | Comments for DerivedClassA |
| [DerivedClassB](classes/DerivedClassB.md) | Comments for DerivedClassB |
| [DisposableClass](classes/DisposableClass.md) | - |
| [DuplicateClass](classes/DuplicateClass-2.md) | Comments for \`DuplicateClass\` |
| [GrandChildClassA](classes/GrandChildClassA.md) | - |
| [GrandChildClassB](classes/GrandChildClassB.md) | - |

## Interfaces

| Interface | Description |
| :------ | :------ |
| [BasicInterface](interfaces/BasicInterface.md) | Comments for BasicInterface |
| [CustomEventInterface](interfaces/CustomEventInterface.md) | - |
| [ExtendedInterface](interfaces/ExtendedInterface.md) | Comments for ExtendedInterface |
| [IndexableInterface](interfaces/IndexableInterface.md) | Comments for IndexableInterface |
| [InterfaceWithComments](interfaces/InterfaceWithComments.md) | Comments for interface over two lines |
| [InterfaceWithEventProperties](interfaces/InterfaceWithEventProperties.md) | - |
| [InterfaceWithFlags](interfaces/InterfaceWithFlags.md) | Comments for InterfaceWithFlags |
| [InterfaceWithTypeParameters](interfaces/InterfaceWithTypeParameters.md) | Comments for InterfaceWithTypeParameters |
| [MultipleIndexableInterface](interfaces/MultipleIndexableInterface.md) | - |

## Type Aliases

| Type alias | Description |
| :------ | :------ |
| [\\_\\_TypeDeclarationWithSpecialCharacters\\_](type-aliases/TypeDeclarationWithSpecialCharacters.md) | Comments for object with special characters |
| [ArrayOfObjectsType](type-aliases/ArrayOfObjectsType.md) | - |
| [ArrayType](type-aliases/ArrayType.md) | Comments for ArrayType |
| [ConditionalType](type-aliases/ConditionalType.md) | Comments for ConditionalType |
| [ExplicitTypeDefOptions](type-aliases/ExplicitTypeDefOptions.md) | - |
| [ExternalReferenceType](type-aliases/ExternalReferenceType.md) | Comments for ExternalReferenceType |
| [FunctionType](type-aliases/FunctionType.md) | Comments for FunctionType |
| [IndexAccessType](type-aliases/IndexAccessType.md) | Comments for IndexAccessType |
| [IntersectionType](type-aliases/IntersectionType.md) | Comments for IntersectionType |
| [LiteralType](type-aliases/LiteralType.md) | Comments for LiteralType |
| [PartialMappedType](type-aliases/PartialMappedType.md) | Comments for PartialMappedType |
| [PrimitiveType](type-aliases/PrimitiveType.md) | Comments for PrimitiveType |
| [PromiseTypeWithObject](type-aliases/PromiseTypeWithObject.md) | Comments for promise type returning object |
| [PromiseTypeWithSymbol](type-aliases/PromiseTypeWithSymbol.md) | Comments for promise type returning symbol |
| [QueryType](type-aliases/QueryType.md) | Comments for query type |
| [ReadonlyMappedType](type-aliases/ReadonlyMappedType.md) | Comments for ReadonlyMapedType |
| [StringLiteralType](type-aliases/StringLiteralType.md) | Comments for StringLiteralType |
| [Stuff](type-aliases/Stuff.md) | - |
| [TupleType](type-aliases/TupleType.md) | Comments for TupleType |
| [TypeDefOptions](type-aliases/TypeDefOptions.md) | - |
| [TypeWithExternalSymbolLinkMapping](type-aliases/TypeWithExternalSymbolLinkMapping.md) | Comments for TypeWithExternalSymbolLinkMapping |
| [TypeWithReturns](type-aliases/TypeWithReturns.md) | Comments for TypeWithReturns |
| [TypeWithTypeParams](type-aliases/TypeWithTypeParams.md) | Comments for TypeWithTypeParams |
| [UnionType](type-aliases/UnionType.md) | Comments for UnionType |
| [UnionTypeWithTemplateStrings](type-aliases/UnionTypeWithTemplateStrings.md) | Union with template strings |
| [UsefulUnionType](type-aliases/UsefulUnionType.md) | Comments for useful UnionType |

## Variables

| Variable | Description |
| :------ | :------ |
| [objectLiteralVariable](variables/objectLiteralVariable.md) | Comments for objectLiteralVariable |
| [objectWithSymbol](variables/objectWithSymbol.md) | Comments variable with symbol |
| [someQuery](variables/someQuery.md) | - |
| [stringVariable](variables/stringVariable.md) | A simple string variable |
| [templateVariableString](variables/templateVariableString.md) | A template variable string |
| [typedef](variables/typedef.md) | - |
| [typeOperatorVariable](variables/typeOperatorVariable.md) | Comments for typeOperatorVariable |

## Functions

| Function | Description |
| :------ | :------ |
| [basicFunction](functions/basicFunction.md) | This is a function that is assigned to a variable. |
| [basicFunctionWithReturns](functions/basicFunctionWithReturns.md) | This is a function that is assigned to a variable. |
| [curriedFunction](functions/curriedFunction.md) | Comments for main curriedFunction |
| [functionReturningAFunction](functions/functionReturningAFunction.md) | Comments for function |
| [functionReturningAnObject](functions/functionReturningAnObject.md) | Comments for function |
| [functionReturningAPromise](functions/functionReturningAPromise.md) | Comments for function |
| [functionReturningAString](functions/functionReturningAString.md) | Comments for function |
| [functionReturningAUnionType](functions/functionReturningAUnionType.md) | Comments for function |
| [functionWithArrayOfOptionalStuff](functions/functionWithArrayOfOptionalStuff.md) | Comments for array of stuff? |
| [functionWithArrayOfStuff](functions/functionWithArrayOfStuff.md) | Comments for array of stuff |
| [functionWithArrayOfUnionStuff](functions/functionWithArrayOfUnionStuff.md) | Comments for array of union stuff |
| [functionWithComplexParams](functions/functionWithComplexParams.md) | Function with function parmas |
| [functionWithDefaultParameters](functions/functionWithDefaultParameters.md) | This is a function with a parameter that has a default value. |
| [functionWithMultipleSignatures](functions/functionWithMultipleSignatures.md) | Main function comment. |
| [functionWithNamedParams](functions/functionWithNamedParams.md) | - |
| [functionWithNestedParameters](functions/functionWithNestedParameters.md) | Some nested params. |
| [functionWithOptionalParameters](functions/functionWithOptionalParameters.md) | This is a function with a parameters. |
| [functionWithRestParams](functions/functionWithRestParams.md) | Function with reset parmas |
| [functionWithTypeParameters](functions/functionWithTypeParameters.md) | Function with type parameters |
| [functionWithUnionParams](functions/functionWithUnionParams.md) | - |
| [tupleTypeFunction](functions/tupleTypeFunction.md) | - |
| [tupleTypeFunctionInlineExplicit](functions/tupleTypeFunctionInlineExplicit.md) | - |
| [tupleTypeFunctionOptional](functions/tupleTypeFunctionOptional.md) | - |
"
`;
